cmd/assetsvc/handler.go

```diff
@@ -90,26 +90,72 @@ func getPageAndSizeParams(req *http.Request) (int, int) {
 	return int(pageNumberInt), int(pageSizeInt)
 }
 
-func getAllChartCategories(namespace, repo string) (apiChartCategoryListResponse, error) {
-	chartCategories, err := manager.getAllChartCategories(namespace, repo)
+func extractDecodedNamespaceAndRepoAndVersionParams(params Params) (string, string, string, []string, []error) {
+	paramErr := []string{}
+
+	namespace, namespaceErr := url.PathUnescape(params["namespace"])
+	repo, repoErr := url.PathUnescape(params["repo"])
+	version, versionErr := url.PathUnescape(params["version"])
+
+	if namespaceErr != nil || repoErr != nil || versionErr != nil {
```

Comment:
the idiomatic thing to do is to early return if there is an error, returning an empty response and the error:

```
namespace, err := url.PathUnescape(params["namespace"])
if err != nil {
  return "", "", "", params["namespace"], err
}
```

(I know you are just grouping errors, but the extra complexity is not necessary)

Reply:
Yes, I kinda agree. It is not a big deal not to display every parameter with decoding errors and the complexity is not necessary. I've just updated the PR with this suggestion, thanks!